sample-succintly/interop/Main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+/*
+  Answers the question, are all primitive int types interoperable? 
+  The same test with primitive float types.
+  Results in build error.
+  
+  Version: 20170910
+*/
+import (
+  "fmt"
+  "reflect"
+)
+
+var (
+  f float32 = 7.0
+  x uint = 7
+  y int = 7
+  z uint32 = 7
+  a uint64 = 7
+  b int32 = 7
+  c int64 = 7
+)
+
+func main() {
+  f2 := 7.0
+  fmt.Printf("%f", f)
+  fmt.Println(reflect.TypeOf(f))
```

Comment:
Im not sure what reflect.TypeOf returns but you can also try to use fmt.Printf("type is %T", yourVar)

Reply:
Yes. They're the same. Just tried reflect.TypeOf instead of %T. Difference is, %T can't be used in fmt.Println().